### DISCUSSION TITLE
How to select in the highlighter "whole line" vs "matching only"

### DISCUSSION THREAD

**Discussion by mlerman** [variar/klogg#357] [Q&A] [answered]
I found in the documentation that it is possible to tell the highlighter to highlight only the matching pattern.
However the documentation doesn't tell how to do that.

For example I want to highlight the text "/+/29594" in the line:
some text here/+/29594 some other text here

So I have composed the pattern : "/\+/\d{5} "
It works to highlight the entire line. But how do I highlight only the matching pattern?

I tried to enclose the pattern in square brackets : [/\+/\d{5} ] but in this case it highlight multiple lines instead.

In most cases I want to highlight the entire lines. In some particular cases I want to highlight a particular number. So I expect to tell the highlighter this information in the pattern itself or with a checkbox. The only checkbox I found is Ignore case.

**Comment by variar** [comment, marked as answer]
Hi, I've made a small video demo:

https://user-images.githubusercontent.com/1620716/124351478-32c95500-dc03-11eb-888d-daed3dadad05.mp4

There is a checkbox in highlighter configuration to enable this mode. When enabled, if regular expression has no explicit capture groups then the whole matching part of lines should be highlighted. If there are explicit capture groups then only parts that are in these groups should be highlighted

**Reply by mlerman** [reply]
Which klogg version I need to use?
I don't see the same dialogs :(

**Reply by variar** [reply]
It has been implemented in 20.12 release -- https://github.com/variar/klogg/releases/tag/v20.12

Alternatively development builds also have this feature and more :)
